### ISSUE TITLE
coalesce function should return value for $value.field

### ISSUE THREAD

**Issue by grzegorz-aniol** [orientechnologies/orientdb#8671]
## Expected behavior  
_coalesce_ function should properly identify not null value referenced as $profile.Name

## Actual behavior  
The function returns last argument(i.e. 'Unknown') even when first argument is not null

![image](https://user-images.githubusercontent.com/22835413/48785721-9702a900-ece5-11e8-821b-fefd63a9e621.png)

## Steps to reproduce  

Run following query on demo database:

```
select @rid, @class, Phone, OrderedId,
  $profile.Name,
  coalesce($profile.Name, 'Unknown')
from Customers
let $profile = out('HasProfile')[0]
```

**Comment by pedroteixeira** [comment]
I noticed that it gives expected result if using this syntax:
``` 
select @rid, @class, Phone, OrderedId, coalesce(out('HasProfile')[0].Name, 'Unknown') from Customers
```

**Comment by luigidellaquila** [comment]
Hi @grzegorz-aniol 

I just pushed a fix for this problem, it will be released with v 3.0.12

Thanks

Luigi
